src/core/server/plugins/plugins_service.ts

```diff
@@ -336,6 +339,18 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
         }
       }
 
+      // validate that OSS plugins do not have required dependencies on X-Pack plugins
+      if (OSS_PATH_REGEX.test(plugin.path)) {
+        for (const id of [...plugin.requiredPlugins, ...plugin.requiredBundles]) {
+          const requiredPlugin = pluginEnableStatuses.get(id);
+          if (requiredPlugin && XPACK_PATH_REGEX.test(requiredPlugin.plugin.path)) {
+            throw new Error(
+              `X-Pack plugin or bundle with id "${id}" is required by OSS plugin "${pluginName}", which is prohibited.`
```

Comment:
Should we extend the error message to suggest making it optional?

Reply:
Done in daebba1a 😄